docs/gangplank/usage.md

```diff
@@ -1,3 +1,239 @@
+
+## Terms
+
+- OpenShift: Red Hat's Kubernetes Platform
+- BuildConfig: a CRD used by OpenShift that builds containers and other build artifacts
+- Unbound Pod: Any instance of Gangplank that is running outside the context of an OpenShift BuildConfig
+
+## Design Idea
+
+Gangplank's core design principle is that containers are the most suitable modern method of orchestrating builds. Gangplank grew out of the various Jenkins libraries and scripts codifying the execution of various versions of COSA. 
+
+Ganplank only knows _how to run COSA_, but running COSA does not require Gangplank. Today it understands how to:
+
+- Run on generic Kubernetes version 1.17+ and OpenShift version 3.11 and 4.x.
+- Be used as an OpenShift BuildConfig
+- Execute locally using podman
+- Read meta.json and Jobspec files
+
+### Menu vs Buffet
+
+Gangplank, with the exception of local podman mode, is intended to run as in the CoreOS Assembler container. Prior to Gangaplnk, a considerable amount of time was spend keeping pipelines, JobSpecs and COSA code-bases in sync. Gangplank seeks to eliminate the mismatch by being part of CoreOS Assembler. Once started, Gangplank will be re-executed as a new pod that has suitable permissions and resources to build a CoreOS variant. When running on OpenShift or Kubernetes, Gangplank will use its pod specification to create worker pods. In other words, Gangplank is tightly coupled to its corresponding COSA container.
+
+The origin pod (the first instance of Gangplank) handles the orchestration of workers by:
+- parsing the environment
+- reading the jobspec
+- creating child worker pod definitions
+- sending work to worker pods and waiting for completion of work
+- life-cycle operations (create/delete/etc) for workers
+
+Previous build systems have used Jenkins Kubernetes plugins for the pod creation and life-cycling of the worker pods. The problem with approach is that each OpenShift/Kubernetes environment would have unique differences that caused pipeline drift. For example, the production pipeline for RHCOS uses a different set of secret names than the development location. 
+
+Gangplank, therefore, evaulates its environment to determine the mode of the build. 
+
+*NOTE: When running in a Kubernetes/OpenShift cluster, Gangplank requires a service account that can read secrets AND create/delete pods.*
+
+## Execution Choices
+
+Gangplank has three execution modes each targeted at a different use-case. 
+
+### OpenShift BuildConfig
+
+Gangplank originally started as an OpenShift BuildConfig custom-build strategy. As a result, Gangplank uses the OpenShift BuildAPI v1 object definition for performing builds. When run as a BuildConfig, Gangplank can perform builds via `oc start-build`. 
+
+The BuildConfig mode is intended for developer and re-build tasks. 
+
+### Unbounded Pods on OpenShift or Kubernetes
+
+Gangplank will execute happily on a generic Kubernetes or OpenShift 3.11 or 4.5+ cluster. 
```

Comment:
Is 4.5+ a hard requirement? Or is it the oldest we've tested with?

Reply:
4.5 is the version that has been tested the most. In theory it should work just fine with 4.2+ cluster with kube-virt; its just never been tested.